src/lib.rs

```diff
@@ -258,6 +258,28 @@ where
         });
     }
 
+    fn on_record(&self, span: &span::Id, values: &span::Record<'_>, ctx: Context<'_, S>) {
+        let Some(span) = ctx.span(span) else {
+            return;
+        };
+        if let Some(f) = self.config.filter {
+            if !PerfettoVisitor::new(f).perfetto {
```

Comment:
This looks like a no-op: `PerfettoVisitor::new()` always sets `perfetto` to `false`. It is only updated via `Visit::record_debug()`, did you intend to call `values.record(&mut visitor)`?

On the other hand, the current `PerfettoFilter::with_filter_by_marker()` defines the filter to allow spans through if _any_ of the fields match the given name: we can no longer know that based on just a subset of the fields here in `on_record()` as far as I understand.

Perhaps you are already covering this case with the `if let Some(extension) = span.extensions_mut().get_mut::<idl::Trace>()` below, as no `idl::Trace` is added to the span extensions if the filter in `on_new_span()` mismatched. In that case, there should be no need to read the filter here?

Reply:
Thanks for catching that -- I had it in my mind that the perfetto field would be set in `new` based on the passed filter, but you're right, we'd need to actually record something with it.

I agree that we don't really need it anyways. We'll have checked the filter on `on_new_span`, and here, we can check if we created the perfetto trace data on the span extensions. I'll put up a PR.

Reply:
Created https://github.com/csmoe/tracing-perfetto/pull/4